docs/examples/networking/nat/vcn.tf

```diff
@@ -0,0 +1,215 @@
+variable "tenancy_ocid" {}
+variable "user_ocid" {}
+variable "fingerprint" {}
+variable "private_key_path" {}
+variable "compartment_ocid" {}
+variable "region" {}
+variable "ssh_public_key" {}
+variable "ssh_private_key" {}
+
+# Choose an Availability Domain
+variable "AD" {
+    default = "1"
+}
+variable "InstanceShape" {
+    default = "VM.Standard1.2"
+}
+
+variable "InstanceOS" {
+    default = "Oracle Linux"
+}
+variable "InstanceOSVersion" {
+    default = "7.4"
+}
+provider "oci" {
+    tenancy_ocid = "${var.tenancy_ocid}"
+    user_ocid = "${var.user_ocid}"
+    fingerprint = "${var.fingerprint}"
+    private_key_path = "${var.private_key_path}"
+    private_key_password = "${var.private_key_password}"
+    region = "${var.region}"
+}
+data "oci_identity_availability_domains" "ADs" {
+    compartment_id = "${var.tenancy_ocid}"
+}
+variable "VCN-CIDR" {
+    default = "10.0.0.0/16"
+}
+
+resource "oci_core_virtual_network" "CoreVCN" {
+    cidr_block = "${var.VCN-CIDR}"
+    compartment_id = "${var.compartment_ocid}"
+    display_name = "mgmt-vcn"
+}
+
+resource "oci_core_internet_gateway" "MgmtIG" {
+    compartment_id = "${var.compartment_ocid}"
+    display_name = "MgmtIG"
+    vcn_id = "${oci_core_virtual_network.CoreVCN.id}"
+}
+
+resource "oci_core_route_table" "MgmtRouteTable" {
+    compartment_id = "${var.compartment_ocid}"
+    vcn_id = "${oci_core_virtual_network.CoreVCN.id}"
+    display_name = "MgmtRouteTable"
+    route_rules {
+        cidr_block = "0.0.0.0/0"
+        network_entity_id = "${oci_core_internet_gateway.MgmtIG.id}"
+    }
+}
+
+resource "oci_core_security_list" "MgmtSecurityList" {
+    compartment_id = "${var.compartment_ocid}"
+    display_name = "MgmtSecurityList"
+    vcn_id = "${oci_core_virtual_network.CoreVCN.id}"
+
+    egress_security_rules = [{
+        protocol = "all"
+        destination = "0.0.0.0/0"
+    }]
+
+    ingress_security_rules = [{
+        tcp_options {
+            "max" = 80
+            "min" = 80
+        }
+        protocol = "6"
+        source = "0.0.0.0/0"
+    },
+    {
+        tcp_options {
+            "max" = 443
+            "min" = 443
+        }
+        protocol = "6"
+        source = "0.0.0.0/0"
+    },
+	{
+        protocol = "all"
+        source = "${var.VCN-CIDR}"
+    },
+    {
+        protocol = "6"
+        source = "0.0.0.0/0"
+        tcp_options {
+            "min" = 22
+            "max" = 22
+        }
+    },
+    {
+        protocol = "1"
+        source = "0.0.0.0/0"
+        icmp_options {
+            "type" = 3
+            "code" = 4
+        }
+    }]
+}
+
+resource "oci_core_subnet" "MgmtSubnet" {
+    availability_domain = "${lookup(data.oci_identity_availability_domains.ADs.availability_domains[var.AD - 1],"name")}"
+    cidr_block = "10.0.0.0/24"
```

Comment:
The VCN CIDR has a variable, but the Subnet CIDRs do not. I would think that if one is a variable then the others should be as well.